### PR TITLE
parbase: Do not Leak TFile Objects

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
`TFile::Open()` returns a `TFile*` and expects the caller to `delete` the object after using it.
Use `std::unique_ptr` to handle this.

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Review by dennisklein — CHANGES_REQUESTED** [review]
Looks good, please add the missing `#include <memory>` and this is a go

File: parbase/FairParRootFileIo.cxx [hunk 1]
```diff
@@ -149,49 +149,41 @@ Bool_t FairParRootFileIo::open(const TList* fnamelist, Option_t* option, const T
...
 
         TListIter keyIter(inputKeys);
+        TKey* inpKey;
         while ((inpKey = static_cast<TKey*>(keyIter.Next()))) {
             TObject* tempObj = inFile->Get(inpKey->GetName());
 
             newParFile->cd();
             tempObj->Write();
         }
-        inFile->Close();
```
Comment (fuhlig1) [inline parbase/FairParRootFileIo.cxx:186]: Why the files are not closed any longer? Currently this is done in the destructor of TFile but it this something one should rely on?

Comment (ChristianTackeGSI) [reply parbase/FairParRootFileIo.cxx:186]: I'd say, that this is the whole purpose of a destructor?

The only real problem I can see: `Close` is a virtual method. Calling a virtual method from a destructor is highly not reommended / possibly doesn't do what you would expect it to do.

But then again: Why do we have a destructor, when it is not cleaning up properly?

@dennisklein any thoughts on this?

Comment (dennisklein) [reply parbase/FairParRootFileIo.cxx:186]: My first thought (before @fuhlig1's question) was to absolutely rely on TFile object destruction to also close the file. But now I am puzzled

@fuhlig1 Can you concretize your concern?

**Review by dennisklein — APPROVED** [review]
thx for adding the header

File: parbase/FairParRootFileIo.cxx (hunk 1 above)
Comment (fuhlig1) [reply parbase/FairParRootFileIo.cxx:186]: Since the file is properly closed in the destructor I am fine with the commit.

**Pull request merged**